script/deployment/MainnetDeploymentScript.s.sol

```diff
@@ -0,0 +1,569 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import "forge-std/Script.sol";
+
+import {ModuleRegistry} from "script/ModuleRegistry.sol";
+
+// Import interfaces:
+
+import {IModule_v1} from "src/modules/base/IModule_v1.sol";
+import {ModuleFactory_v1} from "src/factories/ModuleFactory_v1.sol";
+import {OrchestratorFactory_v1} from "src/factories/OrchestratorFactory_v1.sol";
+import {IInverterBeacon_v1} from "src/proxies/interfaces/IInverterBeacon_v1.sol";
+
+import {Governor_v1} from "@ex/governance/Governor_v1.sol";
+import {FeeManager_v1} from "@ex/fees/FeeManager_v1.sol";
+
+// Import scripts:
+import {DeployAndSetUpInverterBeacon_v1} from
+    "script/proxies/DeployAndSetUpInverterBeacon_v1.s.sol";
+
+contract MainnetDeploymentScript is ModuleRegistry, ProtocolConstants {
+    error BeaconProxyDeploymentFailed();
+
+    // ------------------------------------------------------------------------
+
+    // Beacon Deployment Script
+    DeployAndSetUpInverterBeacon_v1 deployAndSetupInverterBeacon_v1 =
+        new DeployAndSetUpInverterBeacon_v1();
+
+    // TransactionForwarder_v1
+    address forwarder_Proxy;
+    address forwarder_Beacon;
+
+    address governor_Beacon;
+    address governor_Proxy;
+
+    address feeManager_Beacon;
+    address feeManager_Proxy;
+
+    address orchestratorFactory_Implementation;
+    address orchestratorFactory_Beacon;
+    address orchestratorFactory_Proxy;
+
+    address moduleFactory_Implementation;
+    address moduleFactory_Beacon;
+    address moduleFactory_Proxy;
+
+    IModule_v1.Metadata[] initialMetadataRegistration;
+    IInverterBeacon_v1[] initialBeaconRegistration;
+
+    bytes buf_constructorArgs;
+
+    /// @notice Deploys all necessary factories, beacons and implementations
+    /// @return factory The addresses of the fully deployed orchestrator factory. All other addresses should be accessible from this.
+    function run() public virtual returns (address factory) {
+        // TODO: Salted deployments! Check where the commit is and add it.
```

Comment:
What exactly do you mean by that?

Reply:
deterministic address deployment (create2). I think Marvin did it somewhere already, but it needs to be found and ported over